src/librustc_hir/definitions.rs

```diff
@@ -457,7 +443,7 @@ impl Definitions {
         // Create the definition.
         let def_id = LocalDefId { local_def_index: self.table.allocate(key, def_path_hash) };
 
-        assert_eq!(self.def_id_to_node_id.push(node_id), def_id);
+        assert_eq!(self.def_id_to_hir_id.push(None), def_id);
```

Comment:
Not sure why we need these pushes with `None` if we re-populate the whole `def_id_to_hir_id` table below anyway.